components/zcash_address/src/kind/unified.rs

```diff
@@ -411,20 +650,15 @@ pub trait Encoding: private::SealedContainer {
 
 /// Trait for for Unified containers, that exposes the items within them.
 pub trait Container {
-    /// The type of item in this unified container.
-    type Item: SealedItem;
-
-    /// Returns the items contained within this container, sorted in preference order.
-    fn items(&self) -> Vec<Self::Item> {
-        let mut items = self.items_as_parsed().to_vec();
-        // Unstable sorting is fine, because all items are guaranteed by construction
-        // to have distinct typecodes.
-        items.sort_unstable_by(Self::Item::preference_order);
-        items
-    }
+    /// The type of data items in this unified container.
+    type DataItem: SealedDataItem;
 
     /// Returns the items in the order they were parsed from the string encoding.
     ///
     /// This API is for advanced usage; in most cases you should use `Self::items`.
-    fn items_as_parsed(&self) -> &[Self::Item];
+    fn items_as_parsed(&self) -> &[Item<Self::DataItem>];
+
+    /// Returns the revision of the ZIP 316 standard that this unified container
+    /// conforms to.
+    fn revision(&self) -> Revision;
```

Comment:
This trait doesn't seem to have any way to get the metadata items.

Reply:
`items_as_parsed` returns a `Vec<Item<DataItem>>`. The type of data items varies between addresses, IVKs and FVKs, but the type of metadata items does not vary; `Item` is an enum that encompasses both data and metadata items.